compiler-plugin-test/src/test/resources/test-src/sql_script/package_10/sample.bal

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/time;
+import ballerina/persist;
+import ballerina/constraint;
+
+@persist:Entity {
+    key: ["needId"],
+    uniqueConstraints: [["beneficiaryId", "urgency"]],
+    tableName: "MedicalNeeds"
+}
+public type MedicalNeed record {|
+    @persist:AutoIncrement
+    readonly int needId = 1;
+    int beneficiaryId;
+    time:Civil period;
+    @constraint:String {
+        maxLength: 20
+    }
+    string urgency;
+    int quantity;
+    @persist:Relation {reference: ["id"], cascadeDelete: true}
+    Item item?;
+|};
+
+@persist:Entity {key: ["id"]}
+public type Item record {
+    @persist:AutoIncrement
+    readonly int id = 3;
+    @constraint:String {
+        length: 50
+    }
+    string name;
+    @persist:Relation {keyColumns: ["itemId1"], cascadeDelete: true}
+    Item1 item1?;
+};
+
+@persist:Entity {key: ["id"]}
+public type Item1 record {
+    @persist:AutoIncrement
+    readonly int id = 5;
+    @constraint:String {
+        maxLength: 20
+    }
```

Comment:
is it only applicable to string type in SQL?

Reply:
The existing impl supports other sql type such as int, float and decimal. Those are supported to set the length but it seems those purpose are different with  the constraint package's max value purpose. 

As per the MYSQL doc, https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html

- INT(M) indicates the maximum display width.
     Eg:  If we define the int type fields like this, 
     > CREATE TABLE Products(
     >        product_id INT,
     >        weight int(2) ZEROFILL,
     >        review int(3) ZEROFILL,
     >        PRIMARY KEY (product_id)
     >  );
     > 
     > INSERT INTO Products (product_id, weight, review) VALUES (5, 5, 5);
     > SELECT * FROM Products;

    the output is:
    > product_id	weight	review
    > 5	                 05	         005

- FLOAT(M,D) or REAL(M,D) or DOUBLE PRECISION(M,D) or DECIMAL(M,D). Here, (M,D) means than values can be stored with up to M digits in total, of which D digits may be after the decimal point.
  Eg:  If we define the float type fields like this, 
   > CREATE TABLE demo (
   >          f1 FLOAT(2, 1),
   >          f2 FLOAT(4, 3),
   >          f3 FLOAT(7, 5)
   >      );
   > 
   >  INSERT INTO demo(f1,f2,f3)VALUES(1 / 3, 1 / 3, 1 / 3);
   >     
   >  SELECT * FROM demo;

   the output is:
   >  f1	f2	            f3
   >  0.3.  0.333	0.33333